dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -877,6 +881,34 @@ public void download(final List<String> args) {
         }
     }
 
+    /**
+     * Validates that any JSON and/or YAML files are syntactically valid.
+     *
+     * @param args
+     */
+    protected void preValidateLaunchArguments(List<String> args) {
+        if (args.contains("--json")) {
+            String jsonFile = reqVal(args, "--json");
```

Comment:
This will fail if there is no argument after `--json`; I don't think you want to do that here. You should probably call `optVal` and only do the check if the value is not null.

Reply:
Also, you have the string `--json` duplicated; I would create at least a local variable for it, if not a constant at the class level (I realize this string is already hard-coded in several places throughout the file, so the latter may increase the scope of the change too much).

Reply:
I was also concerned about the duplication as well. The problem is that the code diverges depending on if its a workflow or tool, furthermore if its an entry vs local-entry, and for tools in particular, the --json string isn't assigned until the workflow language is determined.
I can make the call to check for valid json syntax after the --json string is assigned for each launch case, but that would mean implementing it in 6 or so cases. Which might make writing tests more difficult if I need to write a test for each case. Any suggestions?